### PR TITLE
Modify network tc according to lwip-2.0.2

### PR DESCRIPTION
**Pull request by HONGCHAEHEE**
server_connect()
itc_net_inet_ntop_p()
tc_net_getsockopt_multicast_ttl_loop_p()
tc_net_getsockopt_multicast_ttl_loop_own_p()
tc_net_getsockopt_multicast_ttl_p()
tc_net_inet_ntop()
 : bugs fixed that cause wrong results.

itc_net_setsockopt_p_reuseport()
tc_net_setsockopt_reuseport_p()
 : commented out as SO_REUSEPORT option is unimplemented.
   It will be added by the next release.

tc_net_getpeername_p1()
tc_net_getpeername_udp_p()
 : wrong tcs removed. getpeername() returns error after close().

tc_net_recv_shutdown_n()
tc_net_recvfrom_tcp_conn_n()
 : expected returns changed.

*** How to Test**
 verified using artik053/tc config

 menuconfig
 Networking Support > LwIP options
 > select "IPv6 support"

 Networking Support > LwIP options > IPv6 support > Socket support
 > select "Enable send timeout socket option"
 "Enable SO_RCVBUF socket option"

 Application Configuration > Examples > TestCase Example > Network TestCase Example
 > select "All" except "dup() api", "ITC dup() api", and "ITC fcntl() api".

 TASH>> network_tc

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [d0e5700847654d96ac346b229febf1f7424dfc57] - Code Rule Check (C++) OK.

File: apps/examples/testcase/le_tc/network/itc_net_inet.c [hunk 1]
```diff
@@ -74,18 +74,19 @@ static void itc_net_inet_aton_n(void)
...
 
 #ifdef CONFIG_NET_IPv4
-	ret = inet_ntop(AF_INET, &in_addr, dst, INET_ADDRSTRLEN);
+	ret = inet_ntop(AF_INET, &in_addr, dst4, INET_ADDRSTRLEN);
 	TC_ASSERT_NEQ("inet_ntop", ret, NULL);
 #endif
 
 #ifdef CONFIG_NET_IPv6
-	ret = inet_ntop(AF_INET6, &in_addr, dst, INET_ADDRSTRLEN);
+	ret = inet_ntop(AF_INET6, &in_addr, dst6, INET6_ADDRSTRLEN);
```
Comment (JeongJunSik) [inline apps/examples/testcase/le_tc/network/itc_net_inet.c]: Is not this the second argument "in6_addr"?

File: apps/examples/testcase/le_tc/network/tc_net_inet.c [hunk 2]
```diff
@@ -102,25 +102,26 @@ static void tc_net_inet_ntoa_p(void)
...
 	TC_ASSERT_NEQ("inet_ntop", ret, NULL);
 
 	/* Failure case: size of destination buffer less than INET_ADDRSTRLEN bytes */
-	ret = inet_ntop(AF_INET, &in_addr, dst, 7);
+	ret = inet_ntop(AF_INET, &in_addr, dst4, 7);
 	TC_ASSERT_EQ("inet_ntop", ret, NULL);
 #endif
 #ifdef CONFIG_NET_IPv6
-	ret = inet_ntop(AF_INET6, &in_addr, dst, INET_ADDRSTRLEN);
+	ret = inet_ntop(AF_INET6, &in_addr, dst6, INET6_ADDRSTRLEN);
```
Comment (JeongJunSik) [inline apps/examples/testcase/le_tc/network/tc_net_inet.c]: also, Is not this the second argument "in6_addr"?

**Comment by seinfra** [comment]
Target : [3417692c9c6a4476c4152f194c9f03a372c8d8ab] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [3417692c9c6a4476c4152f194c9f03a372c8d8ab] - Code Rule Check OK.

File: apps/examples/testcase/le_tc/network/itc_net_setsockopt.c [hunk 3]
```diff
@@ -223,12 +223,17 @@ int itc_net_setsockopt_main(void)
 
 	itc_net_setsockopt_n_invalid_opt_name(fd);
 	itc_net_setsockopt_p_rcvbuf(fd);
-	itc_net_setsockopt_p_reuseport(fd);
+	//itc_net_setsockopt_p_reuseport(fd);
```
Comment (sunghan-chang) [inline apps/examples/testcase/le_tc/network/itc_net_setsockopt.c]: Why do you comment this? I can't find this function. If it is not existed, let's remove this line

File: apps/examples/testcase/le_tc/network/tc_net_setsockopt.c [hunk 4]
```diff
@@ -682,7 +682,7 @@ int net_setsockopt_main(void)
 	//tc_net_setsockopt_ipproto_ip_ip_pktinfo_p(fd);
 	tc_net_setsockopt_keepalive_p(fd);
 	//tc_net_setsockopt_rcvbuf_p(fd);
-	tc_net_setsockopt_reuseport_p(fd);
+	//tc_net_setsockopt_reuseport_p(fd);
```
Comment (sunghan-chang) [inline apps/examples/testcase/le_tc/network/tc_net_setsockopt.c]: This will cause compilation warnig about "set but not used"

**Comment by sunghan-chang** [comment]
@HONGCHAEHEE Here are 3 commits but 1st and 3rd commit have same commit title. Could you merge them or rename it?

**Comment by sunghan-chang** [comment]
@HONGCHAEHEE Additionally you should leave a location where you modify.

File: apps/examples/testcase/le_tc/network/tc_net_inet.c (hunk 2 above)
Comment (HONGCHAEHEE) [reply apps/examples/testcase/le_tc/network/tc_net_inet.c]: You're right. I added "in6_addr" for IPv6.

File: apps/examples/testcase/le_tc/network/itc_net_setsockopt.c (hunk 3 above)
Comment (HONGCHAEHEE) [reply apps/examples/testcase/le_tc/network/itc_net_setsockopt.c]: I commented out that functions.

File: apps/examples/testcase/le_tc/network/tc_net_setsockopt.c (hunk 4 above)
Comment (HONGCHAEHEE) [reply apps/examples/testcase/le_tc/network/tc_net_setsockopt.c]: I commented out that functions.

**Comment by seinfra** [comment]
Target : [dc8f6d5b0dba36456cbd3b6977d16fde46b1cf5a] -

**Comment by seinfra** [comment]
Target : [dc8f6d5b0dba36456cbd3b6977d16fde46b1cf5a] -

**Comment by seinfra** [comment]
Target : [927ad2b2557eb7598de98ba948a6325a6a011e1d] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [927ad2b2557eb7598de98ba948a6325a6a011e1d] - Code Rule Check OK.

**Pull request merged**